beacon/goclient/goclient.go

```diff
@@ -0,0 +1,142 @@
+package goclient
+
+import (
+	"context"
+	client "github.com/attestantio/go-eth2-client"
+	eth2client "github.com/attestantio/go-eth2-client"
+	api "github.com/attestantio/go-eth2-client/api/v1"
+	"github.com/attestantio/go-eth2-client/auto"
+	spec "github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/bloxapp/eth2-key-manager/core"
+	"github.com/bloxapp/ssv/beacon"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/shared/slotutil"
+	"github.com/prysmaticlabs/prysm/shared/timeutils"
+	"github.com/rs/zerolog"
+	"go.uber.org/zap"
+	"time"
+)
+
+type goClient struct {
+	ctx      context.Context
+	logger   *zap.Logger
+	network  core.Network
+	client   client.Service
+	graffiti []byte
+
+	blockChannel     chan spec.SignedBeaconBlock
```

Comment:
@nivBlox where is this channel closed?

Reply:
should running infinity as long the operator running.
will add close in case of failure